src/components/screens/AvailableVehicles.tsx

```diff
@@ -46,6 +47,19 @@ export const AvailableVehicles = ({
     fetchData();
   }, []);
 
+  const getCountArray = useCallback((vehicles: AvailableVehiclesProps[]) => {
+    const vehicleCounts = vehicles.reduce<Record<string, number>>((accumulator, vehicle) => {
+      accumulator[vehicle.name] = (accumulator[vehicle.name] || 0) + 1;
+      return accumulator;
+    }, {});
+
+    return Object.keys(vehicleCounts).map((key) => ({
+      id: key,
+      name: key,
+      count: vehicleCounts[key]
+    }));
+  }, []);
```

Comment:
I have a brief understanding what this part of the code is doing. could you explain it in more datail?

Reply:
This code is basically a function that calculates the number of vehicles with the same key in the `vehicles` array and returns an object.